rust/ql/lib/codeql/rust/dataflow/Ssa.qll

```diff
@@ -338,4 +337,37 @@ module Ssa {
 
     override Location getLocation() { result = this.getBasicBlock().getLocation() }
   }
+
+  /**
+   * An SSA definition inserted at a call that may update the value of a captured
+   * variable. For example, in
+   *
+   * ```rust
+   * fn capture_mut() {
+   *   let mut y = 0;
+   *   (0..5).for_each(|| {
+   *     y += x
+   *   });
+   *   y
+   * }
+   * ```
+   *
+   * a definition for `y` is inserted at the call to `for_each`.
+   */
+  class CapturedCallDefinition extends Definition, SsaImpl::UncertainWriteDefinition {
```

Comment:
Presumably this class was only introduced to improve the `toString` representation? I suggest moving it into `SsaImpl` or making it private.